aggregator/src/binary_utils/job_driver.rs

```diff
@@ -72,28 +77,33 @@ where
         clock: C,
         runtime: R,
         meter: Meter,
+        stopper: Stopper,
         min_job_discovery_delay: Duration,
         max_job_discovery_delay: Duration,
         max_concurrent_job_workers: usize,
         worker_lease_clock_skew_allowance: Duration,
         incomplete_job_acquirer: JobAcquirer,
         job_stepper: JobStepper,
-    ) -> Self {
-        Self {
+    ) -> anyhow::Result<Self> {
+        let max_concurrent_job_workers_u32 = u32::try_from(max_concurrent_job_workers)
+            .context("max_concurrent_job_workers was too large")?;
+        Ok(Self {
             clock,
             runtime,
             meter,
+            stopper,
             min_job_discovery_delay,
             max_job_discovery_delay,
             max_concurrent_job_workers,
+            max_concurrent_job_workers_u32,
```

Comment:
IMO, rather than storing the same value twice, do the conversion at point-of-use. It looks like the `_u32` version is only used in one place, during shutdown, so this should be OK. We can keep an attempted conversion in `new` to validate that the given value can be represented as a `u32`, then unwrap when we do the conversion "for real".